### DISCUSSION TITLE
Sumatra does not remember files state

### DISCUSSION THREAD

**Discussion by Sereton** [sumatrapdfreader/sumatrapdf#3228] [Q&A]
I am using the last version( v3.4.6) in Windows 11, the program works fantastic but I have a small niggling problem: The program remembers opened files and state just fine, **but** if I deactivate "remember opened files" but let "remember state" as true, the program does not remember the state. My settings file is just fine, if I revert to remember opened files the state is also remembered. Have anyone had this issue?

**Comment by GitHubRulesOK** [comment]
SumatraPDF keeps two sections of history that are interrelated

FileStates [
]
SessionData [
]

Session Data is short term tabs status
File States is longer term (RememberOpenedFiles = true) and will also be cleared if switched OFF (EXCEPT favorites will keep their file history until removed.) NOTE even pined documents will be removed from home page only the favorites list entries will be kept ignoring "IsPinned = true" however if remember open files is switched on again then the favorites that are pinned will reappear as pinned !

![image](https://user-images.githubusercontent.com/12754913/212796135-6b689705-403c-47d4-a05c-58242dda53c9.png)


 If you set Remember Opened Files OFF, then subsidiary (RememberStatePerDocument = true) is no longer relevant and "greyed out" in Settings Options and all files re-opened revert to fresh default status.

**Reply by Sereton** [reply]
There is no way to remember the state of each file without remembering the tabs then? That assuming I do not use favorites.

**Reply by GitHubRulesOK** [reply]
the key word is Remember..Files = true or false (not remember different states.)

**Reply by Sereton** [reply]
so the answer is No

**Reply by GitHubRulesOK** [reply]
>remember the state of each file without remembering the tabs then?

pedantically it is not "tab" related those are a subidary consideration and can be OFF but retained
you can remember session files without tab states active so many windows open
but again the master setting is oldest switch of `Remember Opened Files` must be true
